usaspending_api/etl/management/commands/load_table_from_delta.py

```diff
@@ -247,3 +293,219 @@ def handle(self, *args, **options):
+            max_parallel_workers.
+            - Currently it is one-half of the max_parallel_workers -> 8
+        - 8 concurrent JDBC connections will be open by 8 executors; the other 12 will sit idle
+        - 8 tasks will be processed by the 8 active executors
+        - each task may include a batch of multiple file paths from the RDD
+        - each active executor will pull 1 file path from its batch in its task, stream the CSV file from S3 to
+          Postgres using the SQL COPY command. Each COPY command copies ~1000 rows in this case
+        - once copied, it then moves on to the next file path in its task's batch.
+        - once the executor is done with its batch, its task is done, there are no more tasks, it will sit idle
+        - which means the DB could be processing 8 COPY commands concurrently
+
+        Args:
+            spark (SparkSession): the active SparkSession to work within
+            df (DataFrame): the source data, which will be written to CSV files before COPY to Postgres
+            delta_db (str): the Delta Lake database in which to find the Delta table as source of the DataFrame
+            delta_table_name (str): the Delta table used as source of the DataFrame
+            temp_table (str): the name of the temp table (qualified with schema if needed) in the target Postgres DB
+                where the CSV data will be written to with COPY
+            ordered_col_names (List[str]): Ordered list of column names that must match the order of columns in the CSV
+                - The DataFrame should have its columns ordered by this
+                - And the COPY command should provide these cols so that COPY pulls the right data into the right cols
+            keep_csv_files (bool): Whether to prevent overwriting of these temporary CSV files by a subsequent write
+                to the same output path. This will instead put them in a timestamped sub-folder of that output path (
+                which must not already exist or an error will be thrown). Defaults to False. If setting to True,
+                be mindful of cleaning up these preserved files on occasion.
+        """
+        logger = get_jvm_logger(spark)
+        csv_path = f"{CONFIG.SPARK_CSV_S3_PATH}/{delta_db}/{delta_table_name}/"
+        if keep_csv_files:
+            csv_path = (
+                f"{CONFIG.SPARK_CSV_S3_PATH}/temp/{delta_db}/{delta_table_name}/"
+                f"{datetime.strftime(datetime.utcnow(), '%Y%m%d%H%M%S')}/"
+            )
+        s3_bucket_with_csv_path = f"s3a://{spark_s3_bucket_name}/{csv_path}"
+
+        logger.info(f"LOAD: Starting dump of Delta table to temp gzipped CSV files in {s3_bucket_with_csv_path}")
+        df_no_arrays = convert_array_cols_to_string(df, is_postgres_array_format=True, is_for_csv_export=True)
+        df_no_arrays.write.options(
+            maxRecordsPerFile=CONFIG.SPARK_PARTITION_ROWS * 3,  # e.g. 25k - 50k seems a good sweet spot
```

Comment:
Given this seems like a number based from our experience and less with the logic, I'd suggest moving this to the top as a constant if the sweet spot moves around depending on circumstances.